File: Packets/src/main/java/me/ImSpooks/core/packets/collection/other/PacketPingResponse.java
package me.ImSpooks.core.packets.collection.other;

import lombok.Getter;
import me.ImSpooks.core.packets.collection.GlobalPacket;
import me.ImSpooks.core.packets.init.Packet;
import me.ImSpooks.core.packets.init.channels.WrappedInputStream;
import me.ImSpooks.core.packets.init.channels.WrappedOutputStream;

import java.io.IOException;

/**
 * Created by Nick on 17 okt. 2019.
 * Copyright © ImSpooks
 */
@GlobalPacket
public class PacketPingResponse extends Packet {

    @Getter private long serverTime;
    @Getter private long estimatedTime;

    public PacketPingResponse(long serverTime, long clientTime) {
        this.serverTime = serverTime;
        this.estimatedTime = serverTime - clientTime + 1;
    }

    public PacketPingResponse() {
    }

    @Override
    public void send(WrappedOutputStream out) throws IOException {
        out.writeLong(this.serverTime);
        out.writeLong(this.estimatedTime);
    }

    @Override
    public void receive(WrappedInputStream in) throws IOException {
        this.serverTime = in.readLong();
        this.estimatedTime = in.readLong();
    }
}

File: ClientBridge/src/main/php/client/packets/collection/other/PacketPingResponse.php
<?php
/**
 * Created by Nick on 24 okt 2019.
 * Copyright © ImSpooks
 */

namespace client\packets\collection\other;

use client\packets\init\Packet;
use client\packets\init\channels\WrappedInputStream;
use client\packets\init\channels\WrappedOutputStream;

class PacketPingResponse extends Packet {

    /** @var int */ 	private $serverTime;
    /** @var int */ 	private $estimatedTime;

    public function __construct(int $serverTime = 0, int $estimatedTime = 0) {
        $this->serverTime = $serverTime;
        $this->estimatedTime = $estimatedTime;
    }

    public function receive(WrappedInputStream $in) {
        $this->serverTime = $in->readLong();
        $this->estimatedTime = $in->readLong();
    }

    public function send(WrappedOutputStream $out) {
        $out->writeLong($this->serverTime);
        $out->writeLong($this->estimatedTime);
    }

    public function getEstimatedTime(): int {
        return $this->estimatedTime;
    }

    public function getServerTime(): int {
        return $this->serverTime;
    }

}
